templates/site/js/main.js

```diff
@@ -114,129 +49,138 @@ function login(){
+        }
+
+        if(email != sessionEmail && email != null){
+            updateUserEmail(username, email);
+        }
     })
+
 }
 
-function getSessionDetails(){
-    //TODO Remove sessionURL and use sessionURLFragment
-    //TODO Remove nodeURL and use {MAGPIE_URL}
-    const sessionURLFragment = "session";
-    const sessionURL = nodeURL + "/magpie/session"
-    //const testSessionURL = testNodeURL + "/magpie/session"
-    return fetch(sessionURL,{
-        method: "GET",
+function updateUserEmail(username,email){
+    const updateURLFragment = "/magpie/users/" + username;
+
+    fetch(updateURLFragment, {
+        method: "PATCH",
         headers: {
             "Accept": "application/json, text/plain",
             "Content-Type": "application/json"
+        },
+        body:JSON.stringify({
+            "email": email
+        })
+    }).then(response => response.json()).then(json =>{
+            if(json.code && json.code == 200){
+                showSaveCheckmark()
             }
-        }).then(response => { return response.json().then(json =>{
-                return json;
-            })
         })
 }
 
-//User functions
-function getUserDetails(username, cookie){
-    //Use fetch to get user details from user/username endpoint in api
-    //Use details returned to populate fields
-
-    //TODO: When done remove loggedUserUrl and put loggedUserURLFragment for the fetch url.  base url should resolve when run from birdhouse
-    const loggedUserURL = nodeURL + "/magpie/users/" + username;
-    //const loggedUserURL = nodeURL + "/magpie/users/current";
-    //const loggedUserURLFragment = "users/current";
-    return fetch(loggedUserURL,{
-        method: "GET",
+function updateUserPassword(username, password){
+    const updateURLFragment = "/magpie/users/" + username;
+
+    fetch(updateURLFragment, {
+        method: "PATCH",
         headers: {
             "Accept": "application/json, text/plain",
-            "Content-Type": "application/json",
-            "Authorization": cookie, //TODO use Authorization header when request is same-origin
+            "Content-Type": "application/json"
+        },
+        body:JSON.stringify({
+            "password": password
+        })
+    }).then(response => response.json()).then(json =>{
+            if(json.code && json.code == 200){
+                showSaveCheckmark()
             }
-        }).then(response => {return response.json()}).then(json =>{
-            console.log("user details");
-            console.log(json);
-            return json;
         })
 }
 
-function storeUserDetails(username, cookie){
-    getUserDetails(username, cookie).then(json => {
-        json.then(userData => {
-            sessionStorage.setItem("username", userData["user"].user_name);
-            sessionStorage.setItem("email", userData["user"].email);
-        })
-    })
+function deleteUser(){
+    var username;
+    var deleteURLFragment;
 
-}
+    getUserDetails().then(json => {
```

Comment:
It isn't necessary to get the current user here. Just use the `/magpie/users/current` path

Reply:
Switched url fragment to use /magpie/users/current path.